File: Easy/389. Find the Difference/389. Find the Difference.ts
function findTheDifference(s: string, t: string): string {
    let result = 0;

    // XOR all characters in both strings
    for (const char of s) {
        result ^= char.charCodeAt(0);
    }

    for (const char of t) {
        result ^= char.charCodeAt(0);
    }

    // The result will be the ASCII code of the added character
    return String.fromCharCode(result);
}

File: Easy/389. Find the Difference/389. Find the Difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        int result = 0;

        // XOR all characters in both strings
        for (char ch : s.toCharArray()) {
            result ^= ch;
        }

        for (char ch : t.toCharArray()) {
            result ^= ch;
        }

        // The result will be the ASCII code of the added character
        return (char) result;
    }
}
